src/Microsoft.Azure.WebJobs.Host/Loggers/UpdateOutputLogCommand.cs

```diff
@@ -139,7 +139,10 @@ public void Dispose()
                     _synchronizedWriter.Flush();
                 }
 
-                snapshot = _innerWriter.ToString();
+                // Explicitly specify the length. Without this, any writes occurring 
```

Comment:
The synchronized writer -- yes. The underlying `StringBuilder` used by `StringWriter` isn't thread-safe so that guarantees we don't have issues with calling `Append()` from multiple threads.
The lock -- I'm assuming it is still needed. This is the only place it's used, but I'm guessing the original writer wanted to make sure this block was locked. I'd be wary of removing it unless we were sure it wasn't needed.